lib/wallet/src/Cardano/Wallet/DB/Sqlite/Migration/NewSubmissionStore.hs

```diff
@@ -0,0 +1,185 @@
+import Cardano.Wallet.DB.Store.Submissions.Model
+    ( TxLocalSubmissionHistory (TxLocalSubmissionHistory) )
+import Cardano.Wallet.DB.Store.Transactions.Model
+    ( TxRelation (TxRelation), TxSet (TxSet) )
+import Cardano.Wallet.DB.Store.Wallets.Model
+    ( MetasAndSubmissionsHistory, TxWalletsHistory )
+import Cardano.Wallet.DB.Store.Wallets.Store
+    ( mkStoreTxWalletsHistory )
+import Cardano.Wallet.Primitive.Types.RewardAccount
+    ( RewardAccount (..) )
+import Cardano.Wallet.Primitive.Types.TokenPolicy
+    ( TokenName (..), TokenPolicyId (..) )
+import Cardano.Wallet.Primitive.Types.TokenQuantity
+    ( TokenQuantity (..) )
+import Cardano.Wallet.Primitive.Types.Tx.SealedTx
+    ( mockSealedTx )
+import Control.Monad.Logger
+    ( runStderrLoggingT )
+import Control.Monad.Morph
+    ( MFunctor (hoist) )
+import Control.Monad.Trans
+    ( MonadIO (liftIO), MonadTrans (lift) )
+import Control.Monad.Writer
+    ( MonadWriter (tell), execWriter )
+import Crypto.Hash
+    ( hash )
+import Data.ByteString.Char8
+    ( ByteString )
+import Data.DBVar
+    ( Store (..) )
+import Data.Map.Strict
+    ( Map )
+import Data.Time
+    ( getCurrentTime )
+import Database.Persist.Sql
+    ( SqlPersistT, insert_, runMigration )
+import Database.Persist.Sqlite
+    ( runSqlite )
+
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Cardano.Wallet.Primitive.Types.Address as W
+import qualified Cardano.Wallet.Primitive.Types.Coin as W
+import qualified Cardano.Wallet.Primitive.Types.Hash as W
+import qualified Cardano.Wallet.Primitive.Types.Tx.TxMeta as W
+    ( Direction (Outgoing), TxStatus (..) )
+
+createDatabase :: IO ()
+createDatabase = runStderrLoggingT $ runSqlite "before_submission.sqlite" $ do
+    runMigration migrateAll
+    liftIO mkWalletEntity >>= insert_
+    hoist (lift . lift . lift) storeAWallet
+
+mkWalletEntity :: IO Wallet
+mkWalletEntity  = do
+  ct <- getCurrentTime
+  pure $ Wallet
+    { walId = wid "a"
+    , walName = "test"
+    , walCreationTime = ct
+    , walPassphraseLastUpdatedAt = Nothing
+    , walPassphraseScheme = Nothing
+    , walGenesisHash = BlockId $ W.Hash ""
+    , walGenesisStart = ct
+    }
+
+storeAWallet :: SqlPersistT IO ()
+storeAWallet = writeS mkStoreTxWalletsHistory history
+
+history :: TxWalletsHistory
+history = (txSet, txMetaAndSubmissions)
+  where
+    txSet :: TxSet
+    txSet = TxSet $
+      execWriter $ do
+        mkRelation "0" dumbTxRelation
+        mkRelation "1" dumbTxRelation
+        mkRelation "2" dumbTxRelation
+    txMetaAndSubmissions :: Map W.WalletId MetasAndSubmissionsHistory
+    txMetaAndSubmissions =
+      [ ( wid',
+          (txMetaHistory, txLocalSubmissionHistory)
+        )
+      ]
+      where
+        txMetaHistory = TxMetaHistory $
+          execWriter $ do
+            mkRelation "0" $ dumbMetaRelation W.InLedger
+            mkRelation "1" $ dumbMetaRelation W.Pending
+            mkRelation "2" $ dumbMetaRelation W.Expired
```

Comment:
I would probably write it like this:

```suggestion
          Map.fromList
              $ zipWith mkDumbMetaRelation ["0","1","2"] [W.InLedger, W.Pending, W.Expired]
  where
    mkDumbMetaRelation name status = mkRelation name (dumbMetaRelation status)
```

if

```hs
mkRelation :: ByteString -> (TxId -> r) -> (TxId, r)
```

As you know, I tend to avoid `execWriter`. 😅 But there is no need to change the code, it's only temporary anyway — I just wanted to mention it.

Reply:
At least I can do it in ephemeral code. If not, I have to do it in my spare time :-)

Reply:
Have your fun. :-)